test/functional/api/v2/organization_controller_test.rb

```diff
@@ -0,0 +1,243 @@
+require 'test_helper'
+
+class Api::V2::OrganizationsControllerTest < ActionController::TestCase
+  def setup
+    @organization = taxonomies(:organization1)
+    @organization.location_ids = [taxonomies(:location1).id]
+    Rabl.configuration.use_controller_name_as_json_root = false
+  end
+
+  test "should get index" do
+    get :index, { }
+    assert_response :success
+    assert_not_nil assigns(:organizations)
+  end
+
+  test "should show organization" do
+    get :show, { :id => taxonomies(:organization1).to_param }
+    assert_response :success
+    show_response = ActiveSupport::JSON.decode(@response.body)
+    assert !show_response.empty?
+    #assert child nodes are included in response'
+    NODES = ["users", "smart_proxies", "subnets", "compute_resources", "media", "config_templates",
+             "domains", "environments", "hostgroups", "locations", "parameters"].sort
+    NODES.each do |node|
+      assert show_response.keys.include?(node), "'#{node}' child node should be in response but was not"
+    end
+  end
+
+  test "should not create invalid organization" do
+    post :create, { :organization => { :name => "" } }
+    assert_response :unprocessable_entity
+  end
+
+  test "should return an error for no params" do
+    post :create
+    assert_response :unprocessable_entity
+  end
+
+  test "should create valid organization" do
+    post :create, { :organization => { :name => "Test Organization" } }
+    assert_response :success
+    show_response = ActiveSupport::JSON.decode(@response.body)
+    assert !show_response.empty?
+  end
+
+  test "should update organization on if valid is organization" do
+    ignore_types = ["Domain", "Hostgroup", "Environment", "User", "Medium", "Subnet", "SmartProxy", "ConfigTemplate", "ComputeResource", "Realm"]
+    put :update, { :id => @organization.to_param, :organization => { :name => "New Organization", :ignore_types => ignore_types } }
+    assert_equal "New Organization", Organization.find(@organization.id).name
+    assert_response :success
+  end
+
+  test "should not update invalid organization" do
+    put :update, { :id => Organization.first.to_param, :organization => { :name => "" } }
+    assert_response :unprocessable_entity
+  end
+
+  test "should destroy organization if hosts do not use it" do
+    assert_difference('Organization.count', -1) do
+      delete :destroy, { :id => taxonomies(:organization2).to_param }
+    end
+    assert_response :success
+  end
+
+  test "should NOT destroy organization if hosts use it" do
```

Comment:
I'm unsure, but maybe this and the `should destroy` test should just instead be unit tests that ensure we check for hosts? Thoughts?